pkg/noobaaaccount/noobaaaccount.go

```diff
@@ -206,18 +206,17 @@ func RunCreate(cmd *cobra.Command, args []string) {
 	if defaultResource == "" { // if user doesn't provide default resource we will use the default backingstore
 		defaultResource = sys.Name + "-default-backing-store"
 	} 
-	// check that default backing store exists
-	defaultRes := &nbv1.BackingStore{
-		TypeMeta: metav1.TypeMeta{Kind: "BackingStore"},
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      defaultResource,
-			Namespace: options.Namespace,
-		},
-	}
-	if !util.KubeCheck(defaultRes) {
-		log.Fatalf(`❌ Could not get BackingStore %q in namespace %q`,
-			defaultResource, noobaaAccount.Namespace)
+
+	isResourceBackingStore, isResourceNamespaceStore := checkResource(defaultResource)
```

Comment:
I think I would split to 2 functions, and first check if it's backingstore and else check if it's namespacestore. And if he has both with the same name (can it really happen?) we would prefer the backingstore. 
Another option (more work) can be to divide the CRD to defaultBackingStore (like is was originally) and to defaultNSStore and then all of this dual checkup won't be needed - and the user can be specific in what he prefers... WDYT?

Reply:
> I think I would split to 2 functions, and first check if it's backingstore and else check if it's namespacestore. And if he has both with the same name (can it really happen?) we would prefer the backingstore. 

I don't think we should prefer the backingstore as this will change the type of poll the user intended. 
I think we should do PersistentError and let the user handle it. 
for that reason I am not sure we should divide into 2 functions as I will always want to check them both. 
I think it can really happen as they are 2 different kubernetes objects. 

>Another option (more work) can be to divide the CRD to defaultBackingStore (like is was originally) and to defaultNSStore and then all of this dual checkup won't be needed - and the user can be specific in what he prefers... WDYT?

I don't think we should add another resource to the CRD, we don't want to have 2 of them at the same time.
Of course, this can be resolved by using admission control.